frontend/src/components/search/SearchPagination.tsx

```diff
@@ -14,40 +14,61 @@ export enum PaginationPosition {
 interface SearchPaginationProps {
   page: number;
   query: string | null | undefined;
-  total?: number | null;
+  totalPages?: number | null;
   scroll?: boolean;
   totalResults?: string;
   loading?: boolean;
 }
 
 const MAX_SLOTS = 7;
 
+// in addition to handling client side page navigation, this client component handles setting client state for:
+// - total pages of search results
+// - total number of search results
 export default function SearchPagination({
   page,
   query,
-  total = null,
+  totalPages = null,
   scroll = false,
   totalResults = "",
   loading = false,
 }: SearchPaginationProps) {
   const { updateQueryParams } = useSearchParamUpdater();
-  const { updateTotalPages, updateTotalResults } = useContext(QueryContext);
-  const { totalPages } = useContext(QueryContext);
-  // Shows total pages from the query context before it is re-fetched from the API.
-  const pages = total || Number(totalPages);
+  const {
+    updateTotalPages,
+    updateTotalResults,
+    totalPages: totalPagesFromQuery,
+  } = useContext(QueryContext);
+
+  // will re-run on each fetch, as we switch from the Suspended version of the component to the live one
+  useEffect(() => {
+    if (!loading) {
+      updateTotalPages(String(totalPages));
+    }
+  }, [updateTotalPages, totalPages, loading]);
+
+  useEffect(() => {
+    if (!loading) {
+      updateTotalResults(String(totalResults));
+    }
+  }, [updateTotalResults, totalResults, loading]);
 
   const updatePage = (page: number) => {
-    updateTotalPages(String(total));
-    updateTotalResults(totalResults);
```

Comment:
with these functions only being called in `updatePage` the client side pagination state was only being updated on pagination clicks, not in response to the search results updating from the completed API request. With this new setup, the state will be set purely based on the results of the search request whenever a new request is made